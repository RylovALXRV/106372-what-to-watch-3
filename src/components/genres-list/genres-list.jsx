import React from "react";
import {GENRE_LINKS, GENRES} from "../../const";
import {connect} from "react-redux";
import {ActionCreator} from "../../reducer";
import PropTypes from "prop-types";

const GenresList = ({onGenreLinkClick, activeGenreLinkId}) => {
  return (
    <ul className="catalog__genres-list">
      {GENRE_LINKS.map((genreLink, i) => (
        <li
          className={`catalog__genres-item ${i === activeGenreLinkId ? `catalog__genres-item--active` : ``}`}
          key={genreLink}
        >
          <a
            href="#"
            className="catalog__genres-link"
            data-genre={GENRES[i]}
            onClick={(evt) => {
              evt.preventDefault();
              onGenreLinkClick(evt.target.dataset.genre, i);
            }}
          >{genreLink}</a>
        </li>
      ))}
    </ul>
  );
};

GenresList.propTypes = {
  onGenreLinkClick: PropTypes.func.isRequired,
  activeGenreLinkId: PropTypes.number.isRequired,
};

const mapStateToProps = (state) => ({
  movies: state.movies,
  currentGenre: state.currentGenre,
  activeGenreLinkId: state.activeGenreLinkId,
});

const mapDispatchToProps = (dispatch) => ({
  onGenreLinkClick(genre, index) {
    dispatch(ActionCreator.changeActiveGenreLink(index));
    dispatch(ActionCreator.changeGenre(genre));
    dispatch(ActionCreator.filterMovies(genre));
  }
});

export {GenresList};
export default connect(mapStateToProps, mapDispatchToProps)(GenresList);
