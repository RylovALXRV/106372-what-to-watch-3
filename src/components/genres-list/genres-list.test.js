import React from "react";
import renderer from "react-test-renderer";
import {GenresList} from "./genres-list.jsx";

it(`Should GenreList render correctly`, () => {
  const tree = renderer.create(
      <GenresList
        onGenreLinkClick={() => {}}
        activeGenreLinkId={1}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
