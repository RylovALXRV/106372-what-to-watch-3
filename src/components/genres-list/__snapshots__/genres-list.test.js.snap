// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should GenreList render correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="All"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      data-genre="Comedy"
      href="#"
      onClick={[Function]}
    >
      Comedies
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Crime"
      href="#"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Documentary"
      href="#"
      onClick={[Function]}
    >
      Documentary
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Drama"
      href="#"
      onClick={[Function]}
    >
      Dramas
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Horror"
      href="#"
      onClick={[Function]}
    >
      Horror
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="KidsAndFamily"
      href="#"
      onClick={[Function]}
    >
      Kids & Family
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Romance"
      href="#"
      onClick={[Function]}
    >
      Romance
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Sci-Fi"
      href="#"
      onClick={[Function]}
    >
      Sci-Fi
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Thriller"
      href="#"
      onClick={[Function]}
    >
      Thrillers
    </a>
  </li>
</ul>
`;
